android/app/src/main/java/org/fossasia/openevent/core/auth/AuthUtil.kt

```diff
@@ -0,0 +1,128 @@
+package org.fossasia.openevent.core.auth
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.support.annotation.StringRes
+import android.support.v4.app.ActivityCompat
+import android.view.View
+import android.widget.ProgressBar
+import android.widget.Toast
+
+import org.fossasia.openevent.OpenEventApp
+import org.fossasia.openevent.R
+import org.fossasia.openevent.core.main.MainActivity
+import org.fossasia.openevent.data.repository.RealmDataRepository
+import org.fossasia.openevent.common.ConstantStrings
+import org.fossasia.openevent.common.api.JWTUtils
+import org.fossasia.openevent.common.utils.SharedPreferencesUtil
+import org.fossasia.openevent.common.utils.Utils
+
+import java.io.IOException
+
+import okhttp3.Authenticator
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.Route
+import timber.log.Timber
+
+object AuthUtil {
+
+    const val VALID = 0
+    const val EMPTY = 1
+    const val INVALID = 2
+
+    private var authenticator: Authenticator? = null
+
+    @JvmStatic
+    val authorization: String
+        get() = formatToken(token)
+
+    private val token: String?
+        get() = SharedPreferencesUtil.getString(ConstantStrings.TOKEN, null)
```

Comment:
If a token is nullable, authorization may as well

Reply:
So you want the token not null ?

Reply:
No, authorization should be nullable as well